zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -138,11 +147,150 @@ public static Either<ProblemDetail, ProcessInstanceSort> toSearchQuerySort(
     return Either.right(null);
   }
 
+  public static Either<ProblemDetail, UserTaskSort> toUserTaskSearchQuerySort(
+      final List<SearchQuerySortRequest> sorting) {
+
+    if (sorting != null && !sorting.isEmpty()) {
+      final var builder = SortOptionBuilders.userTask();
+
+      for (final SearchQuerySortRequest sort : sorting) {
+        final var field = sort.getField();
+        final var order = sort.getOrder();
+
+        if ("creationTime".equals(field)) {
+          builder.creationDate();
+        } else if ("completionTime".equals(field)) {
+          builder.completionDate();
+        } else {
+          // Return an Either.Left with a ProblemDetail for unknown sortBy field
+          final ProblemDetail problemDetail =
+              ProblemDetail.forStatusAndDetail(
+                  HttpStatus.BAD_REQUEST, "Unknown sortBy field: " + field);
+          return Either.left(problemDetail);
```

Comment:
🔧 How about we collect input validation errors like we do in [RequestMapper.toUserTaskUpdateRequest](https://github.com/camunda/camunda/blob/a7606af681cfc94c24d47393c0978f0cf2ea35ed/zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java#L107-L124)? If we return on the first error directly like you do now, users will only receive this input as invalid. If the `order` is also invalid, they will only learn that in a follow-up request after correcting the field name.